### PR TITLE
Added Controllers article for v9

### PR DESCRIPTION
**Pull request by eshanrnh**

### CODE REVIEW

File: Implementation/Controllers/index-v9.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+For details on using Render MVC Controllers, see the [Controller & Action Selection](../Default-Routing/Controller-Selection/) article.
+
+## Surface Controllers
+
+A SurfaceController is an MVC controller that interacts with the front-end rendering of an UmbracoPage. They can be used for rendering view components and for handling Form data submissions.
+SurfaceControllers are auto-routed which means you don't have to add/create your own routes for these controllers to work.
+
+All implementations of Surface Controllers inherit from the base class: `Umbraco.Cms.Web.Website.Controllers.SurfaceController`.
+
+For details on using Surface Controllers, see the [Surface Controllers](../../Reference/Routing/surface-controllers.md) article.
```
Comment (jonat123) [inline Implementation/Controllers/index-v9.md:34]: This should link to the v9 file :-)

File: Implementation/Controllers/index-v9.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+## Umbraco API Controllers
+
+An Umbraco API Controller is an ASP.NET WebAPI controller that is used for creating REST services. These controllers are auto-routed which means that you don't have to add/create your own routes for these controllers to work.
+
+All implementations of Umbraco API Controllers inherit from the base class: `Umbraco.Cms.Web.Common.Controllers.UmbracoApiController`.
+
+For details on using Umbraco API Controllers, see the [Umbraco API Controllers](../../Reference/Routing/WebApi/index.md) article.
```
Comment (jonat123) [inline Implementation/Controllers/index-v9.md:42]: This should link to the v9 file :-)

File: Implementation/Controllers/index-v9.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+- **WebAPI**
+
+    A base class implementation for authorized auto-routed Umbraco API controllers is inherited from: `Umbraco.Cms.Web.BackOffice.Controllers.UmbracoAuthorizedApiController`. This controller inherits from `Umbraco.Cms.Web.Common.Controllers.UmbracoApiController` it is auto-routed. This controller is also attributed with `Umbraco.Cms.Web.Common.Attributes.IsBackOfficeAttribute` to ensure that it is routed correctly to be authenticated for the backoffice.
+
+    Another common base class implementation for the backoffice is `Umbraco.Cms.Web.BackOffice.Controllers.UmbracoAuthorizedJsonController` which inherits from `Umbraco.Cms.Web.BackOffice.Controllers.UmbracoAuthorizedApiController` but has some special filters applied to it to automatically handle anti-forgery tokens for use with AngularJS in the backoffice.
+
+For more details on Routing requirements, see the [Routing requirements for backoffice authentication](../../Reference/Routing/Authorized/index.md) article.
```
Comment (jonat123) [inline Implementation/Controllers/index-v9.md:62]: This should link to the v9 file :-)

File: Implementation/Default-Routing/Controller-Selection/index-v9.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        }
+    }
+}
+```
+
+Ensure that the controller inherits from the base controller `Umbraco.Cms.Web.Common.Controllers.RenderController`. You can override the `Index` method to perform any customizations of your choice.
+
+## Custom Controller Selection
+
+You can create Custom controllers for different Document Types and Templates. This is termed 'Hijacking Umbraco Routes'. For details on how this process works, see the [Custom MVC Controllers (Umbraco Route Hijacking)](../../../Reference/Routing/custom-controllers.md) article.
```
Comment (jonat123) [inline Implementation/Default-Routing/Controller-Selection/index-v9.md:68]: This should link to the v9 file :-)

**Comment by jonat123** [comment]
@eshanrnh Looks good, i have made a few comments, mostly to the links that should link to the V9 version of the articles mentioned :)

**Comment by eshanrnh** [comment]
> @eshanrnh Looks good, i have made a few comments, mostly to the links that should link to the V9 version of the articles mentioned :)

Argghh, the v9 links. How could I forget that.. 🙄 I have updated them now.. 😀

**Pull request merged**